Persistence of settings across playthroughs via JSON file

Mireille-T opened the pull request:
- Added reading from JSON file (if available) that stores saved settings
- Added writing of setting values to JSON file upon modification in menu
  - `HighlightManager.cs`: `isControllerHighlighted`
  - `XRHardwareController.cs`: `isHapticsEnabled`
  - `UIManager.cs`: `functionalVolume`, `aestheticVolume`, `language`, `subtitleLanguage`, `isFloorVisible`
- Separated music time (i.e. duration of background music played) from rest of player data within GameManager since it does not have to be saved in settings and restored on reboot
- Removed Grip Notice (previously saved in GameManager, not in use anymore) from Player prefab and scripts

Mireille-T committed:
Implemented persistence of settings

- Added reading from JSON file (if available) that stores saved settings
- Added writing of setting values to JSON file upon modification in menu
  - `HighlightManager.cs`: `isControllerHighlighted`
  - `XRHardwareController.cs`: `isHapticsEnabled`
  - `UIManager.cs`: `functionalVolume`, `aestheticVolume`, `language`, `subtitleLanguage`, `isFloorVisible`
- Separated music time (i.e. duration of background music played) from rest of player data within GameManager since it does not have to be saved in settings and restored on reboot
- Removed Grip Notice (previously saved in GameManager, not in use anymore) from Player prefab and scripts

Mireille-T committed:
Create persistent-settings.apk

sonalij3 commented on POINT-VR-Chapter-1/Assets/POINT/InputAssets/Player.prefab:
All the changes are from GripNotice being removed from the Player.prefab

Mireille-T committed:
Removed stray debug statement

sonalij3 committed:
New APK + update in version number in project settings

- EinsteinsSpacetime.apk is the new APK uploaded to the Meta Horizon Store for review
- The old APK's name now reflects that the main difference is that it is the version before the change to Unity 2022 and OpenXR
- the testing APK was removed as the main build reflects the changes

sonalij3 reviewed (approved):
All changes look as concise as we can make them and the objective of the task, to save user setting data onto the headset for persistence, is achieved.
